gmail/src/utils/http.ts

```diff
@@ -0,0 +1,78 @@
+import { State } from "../models/state";
+
+/**
+ * Make a JSON RPC call with the following parameters.
+ */
+export function postJsonRpc(url: string, data = {}, headers = {}) {
+    // Debug, to keep a trace of the number of HTTP request made
+    State.debugIncrementHttpCallCount();
+
+    // Make a valid "Odoo RPC" call
+    data = {
+        id: 0,
+        jsonrpc: "2.0",
+        method: "call",
+        params: data,
+    };
+
+    const options = {
+        method: "post",
+        contentType: "application/json",
+        payload: JSON.stringify(data),
+        headers: headers,
+    };
+
+    try {
+        const response = UrlFetchApp.fetch(url, options);
+
+        const responseCode = response.getResponseCode();
+
+        if (responseCode > 299 || responseCode < 200) {
+            return;
+        }
+
+        const textResponse = response.getContentText("UTF-8");
+        const dictResponse = JSON.parse(textResponse);
+
+        if (!dictResponse.result) {
+            return;
+        }
+
+        return dictResponse.result;
+    } catch {
+        return;
+    }
+}
+
+/**
+ * Make a JSON RPC call with the following parameters.
+ *
+ * Try to first read the response from the cache, if not found,
+ * make the call and cache the response.
+ *
+ * The cache key is based on the URL and the JSON data
+ *
+ * Store the result for 6 hours by default (maximum cache duration)
```

Comment:
Could be nice to explain why we need such a cache mechanism.
I'm guessing it's to avoid making the same enrich request to IAP over and over again?

By the way, did you check if there are any limitations to what you can store into the cache?
I'm guessing it's not unlimited, would be nice to avoid crashes when the cache is "full" (like clean older records maybe?).

Reply:
Yes there're some limitation, the biggest limitation is the "cache life time", which is maximum 6 hours :/
Otherwise it's 250 chars for the key
100KB for the value
Maximum 100 keys stored
See: https://developers.google.com/apps-script/reference/cache/cache

Reply:
It seems that they remove oldest stored keys first when you hit the limit so it should work fine.